env/build.go

```diff
@@ -37,6 +56,26 @@ func matches(k1, k2 string) bool {
 	return k1 == k2
 }
 
+func isNotIncludedForAPI(platformAPI *api.Version) func(k string) bool {
+	if platformAPI.Compare(api.MustParse("0.8")) < 0 {
+		return isNotIncluded
+	}
+
+	return func(k string) bool {
+		return isNotIncluded(k) && isNotIncludedForAssets(k)
+	}
+}
+
+func isNotIncludedForAssets(k string) bool {
+	for _, wk := range BuildEnvIncludeAssets {
```

Comment:
If we do the suggestion here https://github.com/buildpacks/lifecycle/pull/591/files#r625318099 this is where we could check the platform support and return true.